### DISCUSSION TITLE
How to use addIncomingFeature block message?

### DISCUSSION THREAD

**Discussion by ChuLiqiang** [resiprocate/resiprocate#346] [Q&A] [answered]
I want block message use DumFeature in addIncomingFeature。Which enum type should I return in C++ to stop subsequent work? FeatureDoneAndEventDone or ChainDoneAndEventDone?
![image](https://github.com/resiprocate/resiprocate/assets/2747722/a29e32d0-83ec-4417-8372-8c1121b748c3)

![image](https://github.com/resiprocate/resiprocate/assets/2747722/0cf3ee1e-5cd6-41db-9a97-03484b4deb14)

**Comment by sgodin** [comment, marked as answer]
DUM features are chained together (in a known order).  If you return
FeatureDoneAndEventDone then other features in the chain can still run
(with other events).  If you return ChainDoneAndEventDone then any features
after this one will not run.  If you only have one feature in the chain,
then both operate the same way.  If you are blocking a message, then I
believe ChainDoneAndEventDone makes the most sense.

Scott
